### DISCUSSION TITLE
Route generate from nuxt config ?

### DISCUSSION THREAD

**Discussion by Mikaleb** [nuxt/nuxt#18334] [Questions] [answered]
Hello everyone, 

In Nuxt 2 we were able to generate routes from an api like so : 

```js
const dynamicRoutes = () => {
  const routes = axios
    .get('https://xxx.com/wp-json/wp/v2/posts?page=1&per_page=20')
    .then((res) => {
      return res.data.map((post) => `/blog/${post.slug}`)
    })
  return routes

export default {
  generate: {
    routes: dynamicRoutes,
  },
```

But how can we do it in Nuxt 3 ?

**Comment by lucpotage** [comment, marked as answer]
It's under development. Static generation is not yet supported in Nuxt 3. Have a look at this RFC: https://github.com/nuxt/framework/discussions/560

**Reply by fansiyao** [reply]
i really really need you support this function . when will you support it ? Please
